public/react-services/wz-authentication.ts

```diff
@@ -98,4 +104,23 @@ export class WzAuthentication{
       throw error;
     }
   }
+
+  /**
+   * This function returns true only if the user has some police that need be filtered.
+   * Returns false if the user has permission for all agents.
+   * Returns true if the user has no one police for agent:read.
+   * @param policies 
+   * @returns boolean
+   */
+  static userHasAgentsPermissions(policies) {
+    const agentReadPolicies = policies['agent:read'];
+    if (agentReadPolicies) {
+      const allIds = agentReadPolicies['agent:id:*'] == 'allow';      
+      const allGroups = agentReadPolicies['agent:group:*'] == 'allow';
+      const denyAgents = Object.keys(agentReadPolicies).filter(k => !k.includes('*') && agentReadPolicies[k] == 'deny').length;
```

Comment:
Maybe you could use `.some` instead of `.filter`. You don't need the array filtered.

Reply:
yes, great catch tks